services/ingestion/providers/opendota/client.py

```
@@ -0,0 +1,58 @@
+            time.sleep(SLEEP_BETWEEN - delta)
+        self._last_request_ts = time.time()
+
+    def _request(self, path: str) -> dict[str, Any]:
+        self._rate_limit()
+        url = f"{BASE_URL}{path}"
+
+        for attempt in range(1, MAX_RETRIES + 1):
+            try:
+                r = requests.get(url, timeout=TIMEOUT)
+                if r.status_code == 429:
+                    sleep = 2 ** attempt + random.random()
```

Ruff (0.15.2)

[error] Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)